route_guide/route_guide_client.cc

```diff
@@ -23,238 +23,324 @@
+                  // second argument gets the call. You can send requests to
+                  // the server using Write*() family functions. In this
+                  // case we just send a unique request and finish our call
+                  // with WriteLast function.
+                  .ready([](auto&& data, auto& call) {
+                    call.WriteLast(data.point);
+                  })
+                  // .body() callback will be invoked every time the server
+                  // will send the response. First argument in lambda gets
+                  // the context, second - the call, third - the response
+                  // object from the server.
+                  .body([this](auto&& data, auto& call, auto&& response) {
+                    // Do not forget to check if the response from the server
+                    // is valid! If so, consequently we process the response
+                    // from the server.
+                    if (response) {
+                      data.feature.Swap(response.get());
+                      if (!data.feature.has_location()) {
+                        std::cout << "Server returns incomplete feature."
+                                  << std::endl;
+                      }
+                      if (data.feature.name().empty()) {
+                        std::cout << "Found no feature at "
+                                  << data.feature.location().latitude() /
+                                         kCoordFactor_
+                                  << ", "
+                                  << data.feature.location().longitude() /
+                                         kCoordFactor_
+                                  << std::endl;
+                      } else {
+                        std::cout << "Found feature called "
+                                  << data.feature.name() << " at "
+                                  << data.feature.location().latitude() /
+                                         kCoordFactor_
+                                  << ", "
+                                  << data.feature.location().longitude() /
+                                         kCoordFactor_
+                                  << std::endl;
+                      }
+                    }
+                  })
+                  // .finished callback is invoked when the call is finished.
+                  // In this callback we just start our continuation.
+                  .finished([](auto&& data, auto& call, auto&& resp) {
+                    call.Start(grpc::Status::OK);
+                  }));
+    };
   }
 
-  void RecordRoute() {
-    Point point;
-    RouteSummary stats;
+  // ListFeatures() function implements the server streaming RPC. The client
+  // sends Rectangle's message to the server, the server responds with a
+  // sequence of messages(stream) which contain specific address which hits
+  // the Rectangle area. The pattern is quite the same as in GetFeauture().
```

Comment:
Some English suggestions:
```suggestion
  // The `ListFeatures()` function demonstrates a streaming RPC. It is
  // comparable to `GetFeature()`, but instead of sending a single `Point` 
  // the client sends a `Rectangle`. The server responds with a stream of 
  // messages containing addresses that fall inside the Rectangle's area.
```